src/main/java/com/ebay/myriad/scheduler/TaskFactory.java

```diff
@@ -63,6 +64,75 @@ private static String getFileName(String uri) {
             }
         }
 
+        private String getConfigurationUrl() {
+            YarnConfiguration conf = new YarnConfiguration();
+            String httpPolicy = conf.get(YARN_HTTP_POLICY);
+            if (httpPolicy != null && httpPolicy.equals(YARN_HTTP_POLICY_HTTPS_ONLY)) {
+                String address = conf.get(YARN_RESOURCEMANAGER_WEBAPP_HTTPS_ADDRESS);
+                if (address == null || address.isEmpty()) {
+                    address = conf.get(YARN_RESOURCEMANAGER_HOSTNAME) + ":8090";
+                }
+                return "http://" + address + "/conf";
+            } else {
+                String address = conf.get(YARN_RESOURCEMANAGER_WEBAPP_ADDRESS);
+                if (address == null || address.isEmpty()) {
+                    address = conf.get(YARN_RESOURCEMANAGER_HOSTNAME) + ":8088";
+                }
+                return "http://" + address + "/conf";
+            }
+        }
+        private Protos.CommandInfo getCommandInfo() {
+            MyriadExecutorConfiguration myriadExecutorConfiguration = cfg.getMyriadExecutorConfiguration();
+            CommandInfo.Builder commandInfo = CommandInfo.newBuilder();
+            if (myriadExecutorConfiguration.getNodeManagerUri().isPresent()) {
+            /*
+            Overall this is messier than I'd like due to Mesos 1760. We can't let mesos untar the distribution,
+            since it will change the permissions.  Instead we simply download the tarball and execute tar -xvpf.
+            We also pull the config from the resource manager and put them in the conf dir.  This is also why we need
+            frameWorkSuperUser
+            */
+
+                LOGGER.info("Using remote distribution");
+
+                String executorCmdPrefix = "export CAPSULE_CACHE_DIR=`pwd`;echo $CAPSULE_CACHE_DIR; ";
+                if (cfg.getFrameworkUser().isPresent()) {
+                    executorCmdPrefix += "sudo -E -u " + cfg.getFrameworkUser().get() + " -H ";
+                }
+                executorCmdPrefix += "java -Dcapsule.log=verbose -jar ";
+
+                String nmURI = myriadExecutorConfiguration.getNodeManagerUri().get();
+
+                //todo(DarinJ) support other compression, as this is a temp fix for Mesos 1760 may not get to it.
+                String tarCmd = "sudo tar -zxpf " + getFileName(nmURI);
+
+                String chownCmd = "";
+                if (cfg.getFrameworkUser().isPresent()) {
+                    chownCmd += "sudo chown " + cfg.getFrameworkUser().get() + " .";
+                }
+
+                String configCopyCmd = "cp conf " + cfg.getYarnEnvironment().get("YARN_HOME") + "/etc/hadoop/yarn-site.xml";
+
+                String cmd = tarCmd + "&&" + configCopyCmd + "&&" + chownCmd + "&&" + executorCmdPrefix + myriadExecutorConfiguration.getPath();
```

Comment:
Maybe chown then configCopy? Or does it need to be this order? Please document the procedure, at least in comments here.